tests/test_study.py

```diff
@@ -539,6 +539,21 @@ def test_load_study(storage_mode, cache_mode):
         assert created_study.study_id == loaded_study.study_id
 
 
+@pytest.mark.parametrize('storage_mode', STORAGE_MODES)
+@pytest.mark.parametrize('cache_mode', CACHE_MODES)
+def test_delete_study(storage_mode, cache_mode):
+    # type: (str, bool) -> None
+
+    with StorageSupplier(storage_mode, cache_mode) as storage:
+        # Test deleting a non-existing study.
+        with pytest.raises(ValueError):
+            optuna.delete_study("invalid-study-name", storage)
+
+        # Test deleting an existing study.
+        study = optuna.create_study(storage=storage, load_if_exists=False)
+        optuna.delete_study(study.study_name, storage)
```

Comment:
I think that it's better to add the following code to check that the study has been deleted (unless the class of `storage` is `InMemoryStorage`):
```python
with pytest.raises(ValueError):
    optuna.delete_study(study.study_name, storage)
```

Reply:
Can I use `if` statement to skip InMemoryStorage? Some people who like xUnit Test Patterns might say it's an anti-pattern.

Reply:
(xUnit Test Patterns is one of the most famous book which introduce a lot of anti-patterns on unit tests)

> Conditional Test Logic is one factor that makes tests more complicated than they really should be.
> http://xunitpatterns.com/Conditional%20Test%20Logic.html

Reply:
> If you don't mind it, I'll add the assertion in this method. If you mind it, I'll separate test methods.

I think that either is OK.
FYI. We sometimes use `if` for such purpose (e.g., [test_study.py#L507](https://github.com/pfnet/optuna/blob/master/tests/test_study.py#L507)).

Reply:
Thank you for your review. I fixed.

Reply:
Thank you!